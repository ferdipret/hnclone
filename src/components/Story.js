import React from 'react'
import { cleanUrl } from '../utils/index'
import moment from 'moment'

import '../styles/story.css'

function Story({ title, by, url, time, score }) {
  return (
    <div className="story-card-wrapper">
      <a href={url} className="story-card">
        <div>
          <div className="story-card-time">
            {moment.unix(time).format('DD MMMM YYYY')}
          </div>
          <div className="story-card-title">{title}</div>
          <div className="story-card-author">By {by}</div>
        </div>

        <div className="story-card-score">{score}</div>

        <div className="story-card-link">{cleanUrl(url)}</div>
      </a>
    </div>
  )
}

export default Story
