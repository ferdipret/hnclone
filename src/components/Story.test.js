import React from 'react'
import { shallow } from 'enzyme'
import Story from './Story'

describe('Story', () => {
  const story = shallow(<Story />)
  it('renders properly', () => {
    expect(story).toMatchSnapshot()
  })
})
