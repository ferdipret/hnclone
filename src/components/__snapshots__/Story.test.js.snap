// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Story />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="story-card"
        href={undefined}
>
        <div>
                <div
                        className="story-card-time"
                >
                        27/05/2018
                </div>
                <div
                        className="story-card-title"
                />
                <div
                        className="story-card-author"
                >
                        By 
                </div>
        </div>
        <div
                className="story-card-score"
        />
        <div
                className="story-card-link"
        />
</a>,
      "className": "story-card-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <div
                        className="story-card-time"
            >
                        27/05/2018
            </div>
            <div
                        className="story-card-title"
            />
            <div
                        className="story-card-author"
            >
                        By 
            </div>
</div>,
          <div
            className="story-card-score"
/>,
          <div
            className="story-card-link"
/>,
        ],
        "className": "story-card",
        "href": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="story-card-time"
>
                27/05/2018
</div>,
              <div
                className="story-card-title"
/>,
              <div
                className="story-card-author"
>
                By 
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "27/05/2018",
                "className": "story-card-time",
              },
              "ref": null,
              "rendered": "27/05/2018",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "story-card-title",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "By ",
                  undefined,
                ],
                "className": "story-card-author",
              },
              "ref": null,
              "rendered": Array [
                "By ",
                undefined,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "story-card-score",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "story-card-link",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="story-card"
          href={undefined}
>
          <div>
                    <div
                              className="story-card-time"
                    >
                              27/05/2018
                    </div>
                    <div
                              className="story-card-title"
                    />
                    <div
                              className="story-card-author"
                    >
                              By 
                    </div>
          </div>
          <div
                    className="story-card-score"
          />
          <div
                    className="story-card-link"
          />
</a>,
        "className": "story-card-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <div
                            className="story-card-time"
              >
                            27/05/2018
              </div>
              <div
                            className="story-card-title"
              />
              <div
                            className="story-card-author"
              >
                            By 
              </div>
</div>,
            <div
              className="story-card-score"
/>,
            <div
              className="story-card-link"
/>,
          ],
          "className": "story-card",
          "href": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="story-card-time"
>
                  27/05/2018
</div>,
                <div
                  className="story-card-title"
/>,
                <div
                  className="story-card-author"
>
                  By 
</div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "27/05/2018",
                  "className": "story-card-time",
                },
                "ref": null,
                "rendered": "27/05/2018",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "story-card-title",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "By ",
                    undefined,
                  ],
                  "className": "story-card-author",
                },
                "ref": null,
                "rendered": Array [
                  "By ",
                  undefined,
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "story-card-score",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "story-card-link",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
